skillmap/src/lib/workspaceProvider.ts

```diff
@@ -45,13 +45,15 @@ export async function getUserStateAsync(): Promise<UserState> {
     // when auth is enabled.
     let userState = await getLocalUserStateAsync();
 
-    // Read synchronized skillmap state from cloud profile. Fallback to workspace-saved state.
-    const skillmapState = await authClient.getSkillmapStateAsync();
-    userState = {
-        ...userState,
-        mapProgress: skillmapState?.mapProgress ?? userState.mapProgress,
```

Comment:
This was what was copying the local user to the cloud user when we hadn't visited skillmaps before